llpc/patch/llpcPatchBufferOp.cpp

```diff
@@ -1521,31 +1520,53 @@ bool PatchBufferOp::RemoveUsersForInvariantStarts(
 }
 
 // =====================================================================================================================
-// Replace a fat pointer load with the intrinsics required to do the load.
-Value* PatchBufferOp::ReplaceLoad(
-    LoadInst* const pLoadInst) // [in] The load instruction to replace.
+// Replace a fat pointer load or store with the required intrinsics.
+Value* PatchBufferOp::ReplaceLoadStore(
+    bool isLoad,
+    LoadInst* const pLoadInst, // [in] The load instruction to replace.
+    StoreInst* const pStoreInst) // [in] The store instruction to replace.
```

Comment:
Have you considered taking a single argument as `llvm::Instruction` and then checking whether it's a load or store in the function body?

Reply:
I considered this, it would definitely look better as the function signature. However, it allows inserting arbitrary instructions, not only loads and stores. With the current one, the compiler checks if the instruction is actually a `LoadInst` or `StoreInst`.

That’s why I used this signature, I’m not sure which of the two possibilities is better.

Reply:
Makes sense to me. But then you should check which one of the two is not null and don't really need the bool, right? Otherwise callers may mistakenly pass both or use the boolean inconsistently with the pointer arguments.